notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,429 @@
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.3
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+# Architecture
+
+
+## 1. At a “high” level, how will the project deliver each of the needed requirements?
+- **Try to think in terms of some manageable (e.g. 10-20) steps**
+- **List needed intermediate products (e.g. pseudo-code, unit tests, etc.)**
+- **Try to be concise but comprehensive?**
+
+The project is split into two tasks:
+
+The first task will be based on simulation work and will go as follows:
+
+For a causal graph where all Xs are independent:
+ - Ingest data
+ - Manipulate data as needed, the output of this step will be a cleaned dataset
+ - Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ - The cleaned dataset will be used in a function that simulates choice data based on the specified model, resulting in a dataset with simulated choices
+ - The dataset will be used to estimate a choice model and recover the necessary parameters. These parameters can be stored in a dictionary.
+ - The previous two steps will be repeated for N times (say 100) and the parameters from each iteration will be stored in a dictionary entry.
+ - A function that takes in the parameters of the estimated model and results in the percentage of time that the true parameter is recovered.
+
+
+For a realistic causal graph with confounding:
+ - Ingest data
+ - Manipulate data as needed, the output of this step will be a cleaned dataset
+ - Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ - Specify the realistic causal graph with desired confounding structures. The causal graph will be saved as a causalgraphicalmodels object and can be saved as .png file
+ - The cleaned dataset will be used in a function that simulates choice data based on the specified model, resulting in a dataset with simulated choices
+ - The dataset will be used to estimate a choice model and recover the necessary parameters. These parameters can be stored in a dictionary.
+ - The previous two steps will be repeated for N times (say 100) and the parameters from each iteration will be stored in a dictionary entry.
+ - A function that takes in the parameters of the estimated model and results in the percentage of time that the true parameter is recovered.
+
+
+## 2. Description of major components of the system and their relations:
+### 2.1 What should each component do?
+Simulation work:
+Simulation functions will generate data to be used in modeling efforts
+Choice model estimation will try to recover the true parameters of the specified model
+
+### 2.2 What is the interface between each component and each other component?
+Each component(classes) will be written to be as independent as possible from other components. Functions within classes
```

Comment:
In Q2.1, you mention that a simulation object will output data, and that an estimation object will take in that data. This is the interface for the two components (as in the point where the two classes meet) between the two classes. Sorry if it wasn't clear earlier!

Reply:
Also, note there should be some sort of function or method that makes ssure that the data that the simulation object outputs is in some guaranteed form that the estimation object is expecting. For instance, do we have all of the correct columns? Is the data in the correct format (long vs wide format), etc.